extra/nrnivmodl-core.in

```diff
@@ -1,104 +1,121 @@
 
-# Parse args (overriding)
+# parse CLI args
 while getopts "n:m:v:d:i:l:p:hV" OPT; do
     case "$OPT" in
     n)
-        params_MECH_NAME="$OPTARG";;
+        # suffix for mechanism library
+        params_MECHLIB_SUFFIX="$OPTARG";;
     m)
+        # nmodl or mod2c binary to use
         params_NMODL_BINARY="$OPTARG";;
-    d)
-        params_DESTDIR="$OPTARG";;
     i)
+        # extra include flags
         params_INCFLAGS="$OPTARG";;
     l)
+        # extra link flags
         params_LINKFLAGS="$OPTARG";;
     V)
+        # make with verbose
         params_VERBOSE=1;;
     p)
-        PARALLELISM="$OPTARG";;
+        # option for parallel build (with -j)
+        PARALLEL_BUILDS="$OPTARG";;
     h)
         echo "$APP_NAME [options, ...] [mods_path]"
         echo "Options:"
         echo "  -n <name>       The model name, used as a suffix in the shared library"
         echo "  -m <mod2c_bin>  NMODL code generation compiler path"
         echo "  -i <incl_flags> Definitions passed to the compiler, typically '-I dir..'"
         echo "  -l <link_flags> Definitions passed to the linker, typically '-Lx -lylib..'"
-        echo "  -d <dest_dir>   Install to dest_dir. Default: off - Only build in {ARCH}"\
-             "        Consider using this option on a second call. Params are preserved."
         echo "  -V              Verbose: show commands executed by make"
-        echo "  -p <n_procs>    Make parallelism. Default: 4"
+        echo "  -p <n_procs>    Number of parallel builds (Default: $PARALLEL_BUILDS)"
         exit 0;;
     ?)
         exit 1;;
     esac
 done
 
+# consume an option
 shift $(($OPTIND - 1))
 
+# only one mod files directory is supported in neuron and coreneuron
 if [ $# -gt 1 ]; then
     echo "[ERROR] $APP_NAME expects at most one mod dir. See syntax: '$APP_NAME -h' "
     exit 1
 fi
 
-
-# If defined mods dir be in $1
+# if defined mods dir be in $1
 if [ $# -eq 1 ]; then
     # Note: due to bug #712 makefile wont handle mod dir with spaces, so we let it fail here
     params_MODS_PATH=$1
 fi
+
+# warn if no mod files provided
 if [ -d $params_MODS_PATH ]; then
-    # warn if no mod files provided
-    ls $params_MODS_PATH/*.mod || echo "WARNING: No mods found!"
+    ls $params_MODS_PATH/*.mod &>/dev/null || echo "WARNING: No mod files found, compiling default ones only!"
 else
     echo "FATAL: Invalid mods directory: '$params_MODS_PATH'"
     exit 1
 fi
 
-# Output options & populate make_params
-mkdir -p "$(dirname "$SAVE_FILE")"
-echo "# nrnivmodl-core options. Mods location: $params_MODS_PATH" > $SAVE_FILE
+# temporary directory where mod files will be copied
+temp_mod_dir=@CMAKE_HOST_SYSTEM_PROCESSOR@/core/mods
+mkdir -p $temp_mod_dir
 
+# copy mod files with include files. note that ${ROOT_DIR}/share
+# has inbuilt mod files and user provided mod files are in $params_MODS_PATH.
+set +e
+for mod_dir in ${ROOT_DIR}/share/modfile $params_MODS_PATH;
```

Comment:
typically user's won't (or shouldn't) provide new version of inbuilt mod file but `yes` - if you provide modified `hh.mod` then it will override inbuilt one. (it's just directory order in the loop : first inbuilt mod files `${ROOT_DIR}/share/modfile` and then copy users mod files from `$params_MODS_PATH`).